### PR TITLE
Contract Compliance Analysis project

### PR DESCRIPTION
**Pull request by mrgiba**
*Description of changes:*

Add new Contract Compliance Analysis sample

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: samples/contract-compliance-analysis/back-end/app.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python3
+
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (krokoko) [inline samples/contract-compliance-analysis/back-end/app.py:3]: License header should be:

```
#
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
# with the License. A copy of the License is located at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
# and limitations under the License.
#
```

Comment (mrgiba) [reply samples/contract-compliance-analysis/back-end/app.py]: ok, to squash the change into the commit

File: samples/contract-compliance-analysis/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+To deploy this project, follow the instructions available in the README files located at the **back-end** and **front-end** folders, in that sequence.
+
+
+# Content Security Legal Disclaimer
+The sample code; software libraries; command line tools; proofs of concept; templates; or other related technology (including any of the foregoing that are provided by our personnel) is provided to you as AWS Content under the AWS Customer Agreement, or the relevant written agreement between you and AWS (whichever applies). You should not use this AWS Content in your production accounts, or on production or other critical data. You are responsible for testing, securing, and optimizing the AWS Content, such as sample code, as appropriate for production grade use based on your specific quality control practices and standards. Deploying AWS Content may incur AWS charges for creating or using AWS chargeable resources, such as running Amazon EC2 instances or using Amazon S3 storage.
+
+# Operational Metrics Collection
```
Comment (krokoko) [inline samples/contract-compliance-analysis/README.md:46]: Could you please replace this with :  

```
Generative AI CDK Constructs samples may collect anonymous operational metrics, including: the region a construct is deployed, the name and version of the construct deployed, and related information. We may use the metrics to maintain, provide, develop, and improve the constructs and AWS services.
```

File: samples/contract-compliance-analysis/images/contract-analysis-architecture.png
Comment (krokoko) [inline samples/contract-compliance-analysis/images/contract-analysis-architecture.png:1]: could you please provide the original file used to generate those architecture diagrams ?

File: samples/contract-compliance-analysis/back-end/stack/sfn/risk/fn-calculate-risk/requirements.txt
Comment (krokoko) [inline samples/contract-compliance-analysis/back-end/stack/sfn/risk/fn-calculate-risk/requirements.txt:1]: is this needed ? empty file

File: samples/contract-compliance-analysis/back-end/stack/sfn/evaluation/fn-evaluate-clauses/requirements.txt
Comment (krokoko) [inline samples/contract-compliance-analysis/back-end/stack/sfn/evaluation/fn-evaluate-clauses/requirements.txt:1]: is this needed ? empty file

File: samples/contract-compliance-analysis/back-end/stack/lambda/post_jobs_fn/requirements.txt
Comment (krokoko) [inline samples/contract-compliance-analysis/back-end/stack/lambda/post_jobs_fn/requirements.txt:1]: is this needed ? empty file

File: samples/contract-compliance-analysis/back-end/stack/lambda/get_jobs_fn/requirements.txt
Comment (krokoko) [inline samples/contract-compliance-analysis/back-end/stack/lambda/get_jobs_fn/requirements.txt:1]: is this needed ? empty file

File: samples/contract-compliance-analysis/back-end/stack/lambda/get_job_fn/requirements.txt
Comment (krokoko) [inline samples/contract-compliance-analysis/back-end/stack/lambda/get_job_fn/requirements.txt:1]: is this needed ? empty file

File: samples/contract-compliance-analysis/back-end/README.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+With all installed, run this command:
+
+```shell
+$ python3 -V && cdk --version && docker info -f "{{.OperatingSystem}}"
+```
+
+An output similar to the following indicates that all is ok to proceed:
+
+```shell
+Python 3.9.6
```
Comment (krokoko) [inline samples/contract-compliance-analysis/back-end/README.md:42]: the versions mentioned here as part of the output seem to be lower than the versions required above ?

File: samples/contract-compliance-analysis/README.md (hunk 2 above)
Comment (mrgiba) [reply samples/contract-compliance-analysis/README.md:46]: ok, updated

File: samples/contract-compliance-analysis/images/contract-analysis-architecture.png
Comment (mrgiba) [reply samples/contract-compliance-analysis/images/contract-analysis-architecture.png:1]: Shared privately

File: samples/contract-compliance-analysis/back-end/stack/sfn/risk/fn-calculate-risk/requirements.txt
Comment (mrgiba) [reply samples/contract-compliance-analysis/back-end/stack/sfn/risk/fn-calculate-risk/requirements.txt:1]: Leaving it there for easier extension

File: samples/contract-compliance-analysis/back-end/stack/lambda/post_jobs_fn/requirements.txt
Comment (mrgiba) [reply samples/contract-compliance-analysis/back-end/stack/lambda/post_jobs_fn/requirements.txt:1]: Leaving it there for easier extension

File: samples/contract-compliance-analysis/back-end/stack/lambda/get_jobs_fn/requirements.txt
Comment (mrgiba) [reply samples/contract-compliance-analysis/back-end/stack/lambda/get_jobs_fn/requirements.txt:1]: Leaving it there for easier extension

File: samples/contract-compliance-analysis/back-end/stack/lambda/get_job_fn/requirements.txt
Comment (mrgiba) [reply samples/contract-compliance-analysis/back-end/stack/lambda/get_job_fn/requirements.txt:1]: Leaving it there for easier extension

File: samples/contract-compliance-analysis/back-end/README.md (hunk 3 above)
Comment (mrgiba) [reply samples/contract-compliance-analysis/back-end/README.md]: Sure, updated

**Review by krokoko — DISMISSED** [review]
I'll deploy on my end to test the steps in the readme but LGTM

**Review by krokoko — APPROVED** [review]

**Pull request merged**